crates/cheatcodes/defs/src/vm.rs

```diff
@@ -134,6 +144,49 @@ interface Vm {
         bytes stderr;
     }
 
+    /// The result of a `getStateDiff` call.
+    struct AccountAccess {
+        /// What accessed the account.
+        address accessor;
+        /// The account whose storage was accessed.
+        /// It's either the account created, callee or a selfdestruct recipient for CREATE, CALL or SELFDESTRUCT.
+        address account;
+        /// The kind of account access.
+        AccountAccessKind kind;
+        /// If the account is initialized or empty
+        bool initialized;
+        /// The previous account balance of the accessed account.
+        uint256 oldBalance;
+        /// The new account balance of the accessed account. This represents the balance change ignoring any reverts that may have occurred.
+        uint256 newBalance;
+        /// Code of the account deployed by CREATE.
+        bytes deployedCode;
+        /// Value passed along with the account access
+        uint256 value;
+        /// Input data provided to the CREATE or CALL
+        bytes data;
+        /// If this access was in a reverted context
+        bool reverted;
+        /// An ordered list of storage accesses made during an account access operation.
+        StorageAccess[] storageAccesses;
+    }
+
+    /// The storage accessed during an `AccountAccess`.
+    struct StorageAccess {
+        /// The account whose storage was accessed.
+        address account;
```

Comment:
So if address A delegatecalls to B, we'd have:
- `AccountAccess.kind = CALL`
- `AccountAccess.account = B`
- `StorageAccess.account = A`
- The slots all refer to storage in A

Is that correct? Should we make the `AccountAccessKind` more granular so it includes staticcall, delegatecall, and callcode, that way it's explicit instead of implicit what kind of call was made?